### PR TITLE
bugfix for fetching query results

### PR DESCRIPTION
**Pull request by yuyicg**
For the field HasMoreRows in TFetchResultsResponse always return false in hiveserver2, in order to fetch query results data through multiple requests, instead, we check if the value in resultSet is empty to judge if there is more data to be fetched.
This bug has been recorded in https://issues.apache.org/jira/browse/HIVE-11631?jql=project%20%3D%20HIVE%20AND%20resolution%20%3D%20Unresolved%20AND%20text%20~%20%22HasMoreRows%22%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC three year ago and still open.

### CODE REVIEW

File: rows.go [hunk 1]
```diff
@@ -20,9 +20,10 @@ type rowSet struct {
 	columns    []*tcliservice.TColumnDesc
 	columnStrs []string
 
-	offset    int
+        offset    int
```
Comment (wangkuiyi) [inline rows.go]: It seems that the indentation of this line doesn't match the one of the next line?

File: rows.go [hunk 2]
```diff
@@ -20,9 +20,10 @@ type rowSet struct {
...
 	rowSet    *tcliservice.TRowSet
-	hasMore   bool
+
+        // resultSet is column-oriented storage format
```
Comment (wangkuiyi) [inline rows.go]: It seems that the indentation of this line doesn't match the next line.

**Review by wangkuiyi — COMMENTED** [review]
I never dug into `database/sql/dirver`, please @tonyyang-svail review this PR.

It seems to me that the indentations in this change are not consistent. @yuyicg did you configure your editor to invoke goimports for auto-reformat everytime you save a .go file?

File: rows.go (hunk 1 above)
Comment (yuyicg) [reply rows.go]: thanks for the careful reminder, fixed already.

**Review by tonyyang-svail — APPROVED** [review]
LGTM.

**Pull request merged**